### DISCUSSION TITLE
Enabling MQTT or reboot with MQTT enabled -> deactivte the  SM solar system

### DISCUSSION THREAD

**Discussion by Helmut1965** [emsesp/EMS-ESP32#2228] [Q&A]
EMS Version: 3.7.0
Board Profile: BBQKees Gateway E32, V1.5
TX Mode: EMS+, Service Key 0x0B

Devices: GB192, RC310, Solar Module SM100, WaterModule MS100

Problem Description: During the last day I try to change from the IOBroker EMS adapter to an mqtt communication. To do so, I enable MQTT in EMS-ESP. After I enable the MQTT it deactivates a few minutes later the Solar System. Modul is still available.

The entity 'activated' is set to off. I'm not able to enable this in web interface, I have to renable it in the RC310 at the device.
Also in case I reboot the EMS-ESP or I disconnet the ethernet cable, the Solar system will be disabled.
I copied a few log entries which was created during the enablement of the MQTT. 
The problem is also there in case I change from ESP+ to ESP.

[log.txt](https://github.com/user-attachments/files/17813691/log.txt)

Is there something else what I can provide?
Have a great Day 
Helmut

**Comment by proddy** [comment]
looking at the logs I see something is sending many MQTT commands to EMS-ESP. Do you know what this could be?

**Comment by MichaelDvP** [comment]
There is no system info, i need to use my foggy crystall ball. 
In this i see emsesp-mqtt-options `publish-single` and `publish-to-command` are checked and ioBroker adapter is set to `publish own states` and/or `publish on subscribe`. So we have a nice publish-command ping-pong-game without winners.

**Reply by Helmut1965** [reply]
EMS-ESP configuration is: Topic/Payload => As individual topic
Publish command output to a `response` topic => not checked
Publish single value topics on change => checked
Publish to command topics (ioBroker) => not checked
Set Clean Session => checked
Always set Retain flag => not checked

Publish intervall all to 240s
QoS = 0

**Reply by Helmut1965** [reply]
To add in addition: IOBroker use a MQTT dlient to read or publish, currently only read. The broker is running as a standalone broker, there is nothing configured as publish own states. Easy to check with the topic '#' on the broker which shows all. 
This Mqtt messages in the logs comes only during boot or enable MQTT they are gone during a minute or so.
Overall, the problem are not these messages, the problem is - enabling MQTT deactivates my solar system.

**Reply by MichaelDvP** [reply]
> Overall, the problem are not these messages, the problem is - enabling MQTT deactivates my solar system.

The problem are these messages. Your mqtt broker sends a lot of commands to emsesp, one of  them is probably: solar/enable false.
The error messages are generated before solar is registered. At this time emsesp is also not subscribed to solar messages, the broker should not send any command for solar at this time. Also emsesp never subscribe to `dallassensor`, 
There is something seriously wrong in your mqtt-broker config. 
Try to use mqtt explorer and remove all entries  for `emsesp/solar/` and `emsesp/dallassensor` and `emsesp/<device>` if there are anyother devices. You can leave the `emsesp/<device>_data/` entries.
But still not clear why the broker sends topics that are not suscribed.

**Reply by Helmut1965** [reply]
Dear MichaelDvP, 
thanks for the reply. I did a few checks on this. Overall, there is nothing which send any commands to the ESP ... But ... During my first test I started the EMS-ESP Mqtt with the retain flag. Based on this all messages from the EMS-ESP are stored on the Broker, in case the EMS-ESP did a restart for which ever reason it get all his own messages from the past back 
After I realized this, I restart my MQTT Broker (retain no) and the messages are gone. After this, the logs are clean and the EMS is there without deactivating the solar part. Overall, here is one miracle: How disabled the Solarsystem on the first MQTT Client start on the EMS which caused the retain message. I will monitor this to see what will happen here over time.
In principle, this can happen at any time in case somebody runs an environment with retain, maybe it make sense to add something to the documentation that this messages can be ignored.

There is also something what I can get for the dallassensor. This is caused by the EMS-ESP adapter instance on the IOBroker. I checked the box in the adapter that dallas temp sensor should be checked. In case I uncheck this, the message is gone as well. Unfortunately also the temp sensors connected to EMS-ESP box are not recorded anymore. I think I have a few of them which works great, they are maybe not a dallassensor. I think this is an issue in the IOBroker adapter.

During all the test something happen which remind me to another Bug report. I opened the web interface for a longer time and in addition the IOBroker connects every 240 seconds over the WEB API. After some time, the device frozen completely. My switch reported - still connected, access with ping and telnet isn't possible anymore. This happened around 3h. I reboot the device and closed the browser window. Currently it runs stable for around 20hours.

Thanks for looking into my topic and providing some good hints.
Helmut

**Reply by MichaelDvP** [reply]
What ems-esp versio do you use, what ioBroker adapter version?
With actual versions the temperaturesensors are shown without any options checked in the adapter.

**Reply by Helmut1965** [reply]
@MichaelDvP 
1) EMS-ESP: 3.7.0
2) IOBroker Adapter 4.8.0
3) There are no Tempsensors in the Adapter in case I deselect the Dalas checkbox

**Reply by MichaelDvP** [reply]
Ok, the better temperaturesensor came with Adapter 4.9 see: https://github.com/tp1de/ioBroker.ems-esp/releases/tag/v4.9.0
Ems-esp 3.7.0. has the v2-api bug, that could make it freeze, i can't remember, has the Adapter 4.8 the v2-api option or use it only v3 protocol? 
Try update both, you always can go back if you don't like it.

**Reply by Helmut1965** [reply]
@MichaelDvP , thanks for reply. My target was MQTT and get rid of the EMS-ESP Adapter. The MQTT is faster because it send state changes on his own and not polling.  Also it's great, that there is the possibility ton enable or disable the entities in the web interface for the API/MQTT. 
There is maybe one small improvement idea, It would be great to enable/disable an entity on change but leave in the status for example every 600s. To give one example. My syspress is send every few seconds because it is between 1.3 and 1.4. So value is update very frequently. In case I disable this for sending - it's not longer there, also away from the status (default cycle). It would be great, if there is the possibility to leave all items on the status every n-seconds.
I hope, I was able to explain my thought here.

**Reply by MichaelDvP** [reply]
Yes, publish on change can only used with mqtt, API is alwas a response.
There are two possibilities for publish on change, setting publish interval to zero publishes the json of all device values on any change within this json. Checking `Publish single values on change` publishes additional small messages for each entity seperate. Publish is always when a ems-telegram comes in checking the values for changes and publish. So publish time is given by the bus and we don't have too much publishes at one time. Setting these single values to timing will blow the message queue. Using `publish single values` and setting interval to zero disables the json publish.
Works with any mqtt-broker like mosquitto, etc., In ioBroker looks like this:
![grafik](https://github.com/user-attachments/assets/dd1a99c0-02e2-450b-ae6d-5277b6ef9626)

There is another option: `Publish to command topic (ioBroker)` in mqtt-settings. This can only be used with the ioBroker-mqtt https://github.com/ioBroker/ioBroker.mqtt as mqtt-broker with disabled settings for publish on subscribe and publish on connect.
The ems-esp publishes to to the same topic used for commands, e.g. `thermostat/hc1/seltemp` instead of `thermostat_data/hc1/seltemp`. A standalone mqtt-broker as mosquitto will send any message back to to the subscription of ems-esp, so any publish of emsesp results in a write command, that's bad. The ioBroker mqtt just collects the data and sends a command only if you change in the integrated ioBroker-client. These entitiescan be used direct in ioBroker-Vis to change values. Looks like this:
![grafik](https://github.com/user-attachments/assets/78259e51-f729-4141-9948-a00ddebfcb6a)
But with publish on subscribe/connect in ioBroker-mqtt or when using mosquitto this leads to unwanted commands like you have with the deactivated solar.
